src/dataServices/apiDataService.ts

```diff
@@ -0,0 +1,116 @@
+/* eslint-disable class-methods-use-this */
+
+import {
+    ConditionalDeleteResourceRequest,
+    CreateResourceRequest,
+    DeleteResourceRequest,
+    GenericResponse,
+    PatchResourceRequest,
+    Persistence,
+    ReadResourceRequest,
+    ResourceNotFoundError,
+    UpdateResourceRequest,
+    vReadResourceRequest,
+} from 'fhir-works-on-aws-interface';
+import axios from 'axios';
+
+export class ApiDataService implements Persistence {
+    updateCreateSupported: boolean = false;
+
+    // TODO: Pull this value from AWS Param Store
+    static readonly INTEGRATION_TRANSFORM_URL = 'http://localhost:4000';
+
+    async createResource(request: CreateResourceRequest): Promise<GenericResponse> {
+        try {
+            const url = `${ApiDataService.INTEGRATION_TRANSFORM_URL}/${request.resourceType}`;
+            const response = await axios.post(url, request.resource);
+            return { message: '', resource: response.data.resource };
+        } catch (e) {
+            throw this.getError(e);
+        }
+    }
+
+    async readResource(request: ReadResourceRequest): Promise<GenericResponse> {
+        try {
+            const response = await axios.get(
+                `${ApiDataService.INTEGRATION_TRANSFORM_URL}/${request.resourceType}/${request.id}`,
+            );
+            return { message: '', resource: response.data.resource };
+        } catch (e) {
+            throw this.getError(e, request.resourceType, request.id);
+        }
+    }
+
+    async updateResource(request: UpdateResourceRequest): Promise<GenericResponse> {
+        try {
+            const response = await axios.put(
+                `${ApiDataService.INTEGRATION_TRANSFORM_URL}/${request.resourceType}/${request.id}`,
+                request.resource,
+            );
+            return { message: '', resource: response.data.resource };
+        } catch (e) {
+            throw this.getError(e, request.resourceType, request.id);
+        }
+    }
+
+    async deleteResource(request: DeleteResourceRequest): Promise<GenericResponse> {
+        try {
+            await axios.delete(`${ApiDataService.INTEGRATION_TRANSFORM_URL}/${request.resourceType}/${request.id}`);
+            // Don't need to actually return anything to the router
+            return { message: '', resource: {} };
+        } catch (e) {
+            return this.getError(e, request.resourceType, request.id);
+        }
+    }
+
+    getError(e: any, resourceType: string = '', id: string = '') {
```

Comment:
when calling axios.post() if that responds in a non-200 does that throw an error?

Reply:
Yep, non-200 response will result in an error being thrown by axios.